prometheus/base/alerting/prometheus.argocd_alerts.yaml

```diff
@@ -17,3 +17,14 @@ spec:
       annotations:
         summary: "Application {{ $labels.name }} in namespace {{ $labels.dest_namespace }} was 'Degraded'"
         description: "Application {{ $labels.name }} in namespace {{ $labels.dest_namespace }} was 'Degraded' in the last 5 minutes"
+    - alert: ProgressingArgocdApp
+      expr: |
+        (avg_over_time(argocd_app_info{health_status="Progressing", dest_namespace!~".+-tenant"}[20m]) > 0.5)
```

Comment:
Since the values are not `1` or `0`, but `1` or `_`,  we saw that the use of `avg_over_time` is giving us wrong values.
The average will always be `1` even if it appears only once, in this series for example : `_x16 1 _ _`, we have 19 `_` and only one `1`, and the average is 1 since the `_` are not accounted as `0`

We have to come up with different approach

Reply:
It's changed, the rule now consists of a check whether it was "Progressing" in the last 20 minutes and also if it's not "Healthy" for the past 4 minutes.

Reply:
Didn't we want to check if  `Progressing` > `Healthy` in the past 20 minutes?

Reply:
@ShakedAviv50313 found a bug there. If there are no healthy readings inside the time frame, then the comparison will fail.

i.e. to have any operator between 2 series, we need each of them to have at least one reading within the time frame, otherwise the operation will not return any time series. This means that that option will not detect an app which is progressing throughout the time window (or flapping between progressing and anything other than healthy).

Reply:
In this case we can use the `or vector(0)` which will return a vector of `0`s in case the query returns None or null:

```
(count_over_time(argocd_app_info{health_status="Progressing", dest_namespace!~".+-tenant"}[20m] or vector(0)) 
> ignoring (health_status) 
count_over_time(argocd_app_info{health_status="Healthy", dest_namespace!~".+-tenant"}[20m]) or vector(0)) 
and ignoring (health_status) 
max_over_time(argocd_app_info{health_status!="Healthy", dest_namespace!~".+-tenant"}[4m]) == 1
```

I haven't test it yet, so it might need a little tweaking, but I think that that's the approach

Reply:
Indeed needs quite some testing but that might be a way forward here.

Reply:
It passes all the tests!! so i'm changing this pr for the new rule, just for general knowledge i was looking at and trying to use the `absent(sample_query)` which return 1 if a query does not return anything, and 0 otherwise.

Reply:
@avi-biton @yftacherzog can I please get a second review on this?